jerry-core/api/jerry.c

```diff
@@ -1710,10 +1711,37 @@ jerry_create_regexp_sz (const jerry_char_t *pattern_p, /**< zero-terminated UTF-
   }
 
   ecma_string_t *ecma_pattern = ecma_new_ecma_string_from_utf8 (pattern_p, pattern_size);
+  ecma_stringbuilder_t builder = ecma_stringbuilder_create ();
 
-  jerry_value_t ret_val = ecma_op_create_regexp_object (ecma_pattern, flags);
+  if (flags & RE_FLAG_GLOBAL)
+  {
+    ecma_stringbuilder_append_char (&builder, LIT_CHAR_LOWERCASE_G);
+  }
+  if (flags & RE_FLAG_IGNORE_CASE)
+  {
+    ecma_stringbuilder_append_char (&builder, LIT_CHAR_LOWERCASE_I);
+  }
+  if (flags & RE_FLAG_MULTILINE)
+  {
+    ecma_stringbuilder_append_char (&builder, LIT_CHAR_LOWERCASE_M);
+  }
+  if (flags & RE_FLAG_STICKY)
+  {
+    ecma_stringbuilder_append_char (&builder, LIT_CHAR_LOWERCASE_S);
+  }
+  if (flags & RE_FLAG_UNICODE)
+  {
+    ecma_stringbuilder_append_char (&builder, LIT_CHAR_LOWERCASE_U);
+  }
+
+  ecma_value_t flags_value = ecma_make_string_value (ecma_stringbuilder_finalize (&builder));
```

Comment:
I'd really prefer if we didn't create this string just so that we can re-parse it later. There should be a way to create a regexp with already parsed flags.